responsemanager/server.go

```diff
@@ -227,7 +275,9 @@ func (rm *ResponseManager) startTask(task *peertask.Task) queryexecutor.Response
 
 func (rm *ResponseManager) finishTask(task *peertask.Task, err error) {
 	key := task.Topic.(responseKey)
-	rm.responseQueue.TaskDone(key.p, task)
+	defer func() {
+		rm.responseQueue.TaskDone(key.p, task)
```

Comment:
Why are we putting this in defer here?

Reply:
I don't recall! This was an early change, I suspect it comes back to the problem of waiting for all work to complete before collecting traces in the tests - any span that's not closed by the time we collect the traces won't show up, hence all the sync/drain stuff I've had to add. This `defer` may be unnecessary now with the more recent `drain()` functionality.